Tools/ApplicationDebugger/jacobi/src/jacobi-fixed.cpp

```diff
@@ -0,0 +1,119 @@
+//==============================================================
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+// The program solves the linear equation Ax=b, where matrix A is a
+// n x n sparse matrix with diagonals [1 1 4 1 1],
+// vector b is set such that the solution is [1 1 ... 1]^T.
+// The linear system is solved via Jacobi iteration.
+
+// Each call of the kernel computes one iteration of the Jacobi method.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+// Location of file: <oneapi-root>/dev-utilities/<version>/include
+#include "dpc_common.hpp"
+#include "selector.hpp"
+
+using namespace std;
+using namespace sycl;
+
+int main(int argc, char *argv[]) {
+  constexpr size_t n = 64;
+  int b[n];
+  float x_k[n];
+  float x_k1[n];
+
+  // Initialize the input.
+  for (int i = 0; i < n; i++) {
+    b[i] = 8;
+    x_k[i] = 0;
+  }
+
+  b[0] = 6;
+  b[1] = 7;
+  b[n - 2] = 7;
+  b[n - 1] = 6;
+
+  try {
+    CustomSelector selector(GetDeviceType(argc, argv));
+    queue q(selector, dpc_common::exception_handler);
+    cout << "[SYCL] Using device: ["
+         << q.get_device().get_info<info::device::name>() << "] from ["
+         << q.get_device().get_platform().get_info<info::platform::name>()
+         << "]\n";
+
+    // Create 1D buffers for b vector and iteration vectors.
+    buffer buffer_b(b, range{n});
+    buffer buffer_x_k(x_k, range{n});
+    buffer buffer_x_k1(x_k1, range{n});
+
+    for (int k = 0; k < 15000; k++) {
```

Comment:
1500 is an arbitrary constant which would benefit from using a named constexpr.    Also for debugging perspective do you need this many iterations, or would something much smaller like 10 be enough to demo this?

Reply:
With my original matrix the convergence rate was indeed too low and required ~15000 iterations, as the matrix was not strictly diagonally dominant. I changed the matrix to [1 1 5 1 1], now the diagonal dominates (5 > 1+1+1+1) and the method converges in 100 iterations.
Also made the number of iterations a constexpr.
Thank you for this comment!